hack/setup-dev.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+root=$(dirname "${BASH_SOURCE[0]}")
+
+bash -e $root/install-dev-deps.sh
+source $root/dev-functions.sh
+
+# Create the control cluster
+kind delete cluster --name argocd-control-plane
+kind create cluster --name argocd-control-plane
+kubectl create namespace argocd
+
+# Setup argocd
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
+kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj-labs/applicationset/v0.1.0/manifests/install.yaml
+
+# Login as admin and change def pass
+sleep 10
```

Comment:
what is the sleep step for?

Reply:
Sometimes the object is not immediately retrievable in kubernetes so this makes sure that the watch on the line below doesn't fail because the object has not yet been created.

Reply:
i've been there :-) could this be simplified / clarified by a `kubectl wait`?